### ISSUE TITLE
Improve JSX components integration

### ISSUE THREAD

**Issue by konstantinmuenster** [mdx-editor/editor#20]
Hey and thanks for the awesome library. Really blown away by how well it works out of the box! 

I tested it with a few of my blog posts and I noticed 2 things that could be improved for the JSX components integration.

1. A toolbar button to add a new component (e.g. a dropdown that shows you all the components you configured via `jsxComponentDescriptors`)
2. If the component has nested text as children, make the hierarchy visible in the editor (e.g. by nesting the text or adding a background).

An example for the 2nd issue:

I have this component in Markdown:

```
<Callout>
  The principles are applicable to all areas of programming, even though the
  title says ‘frontend developers’ specifically.
</Callout>
```

and it resolves correctly in the editor. But you can't tell whether or not the text belongs to the `Callout` editor.

<img width="687" alt="Bildschirm­foto 2023-07-20 um 06 45 39" src="https://github.com/mdx-editor/editor/assets/46243719/1b77528c-05d9-440a-ad59-0514f45c0f56">

**Comment by konstantinmuenster** [comment]
@petyosi since you also asked for exemplary JSX components on Discord, you can have a look at the components I use for my blog: https://github.com/konstantinmuenster/konstantin.digital/tree/main/app/blog/%5Bslug%5D/_mdxComponents

Hope it helps to identify common usecases 🙌

**Comment by petyosi** [comment]
Thank you! Those components are super helpful. I'm currently working on letting the developers pass (easily) custom components as editors for markdown directives. Here's an example of a [simple youtube directive](https://github.com/mdx-editor/editor/blob/leaf-directive/src/stories/youtube.stories.tsx#L23-L49).

I believe that the same concept can make it easy to bring a first-class editing experience for the JSX components, too.

**Comment by konstantinmuenster** [comment]
Oh, that looks cool! So, in my case of the Callout component, you could render a styled container like this, right?

```tsx
<div style={{ backgroundColor: 'red' }}>
    {mdastNode.children}
</div>
```

**Comment by petyosi** [comment]
More or less - yes. You would be able to specify the editor in the JsxComponents config. A bit more would be necessary to make the contents editable with a nested editor, but based on your feedback, I am going to make that convenient enough.

**Comment by petyosi** [comment]
@konstantinmuenster - Utilities for custom JSX editors landed in v0.11.0. Take a look at [this example](https://mdxeditor.dev/editor/docs/jsx).
